File: app/src/main/java/io/mgba/Utils/IDependencyInjector.java
package io.mgba.Utils;

import io.mgba.Model.Library;
import io.mgba.Presenter.IntroPresenter;
import io.mgba.Presenter.MainPresenter;

public interface IDependencyInjector {
    void inject(IntroPresenter target);
    void inject(MainPresenter target);
    void inject(Library library);

}

File: app/src/main/java/io/mgba/utilities/IDependencyInjector.kt
package io.mgba.utilities

import io.mgba.model.Library
import io.mgba.presenter.IntroPresenter
import io.mgba.presenter.MainPresenter

interface IDependencyInjector {
    fun inject(target: IntroPresenter)
    fun inject(target: MainPresenter)
    fun inject(library: Library)

}
